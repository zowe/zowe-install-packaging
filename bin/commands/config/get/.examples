zwe config get -p path-of-configuration -c /path/to/zowe.yaml
zwe config get -p .zowe.certificate -c /path/to/zowe.yaml
zwe config get -p .zowe.components[] -c /path/to/zowe.yaml
zwe config get -p .zowe.components.gateway -c /path/to/zowe.yaml
zwe config get -p .zowe.components.gateway.enabled -c /path/to/zowe.yaml