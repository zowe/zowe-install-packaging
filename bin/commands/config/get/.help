Return value of a configuration defined in YAML configuration.
This command requires zowe.useConfigmgr=true or --configmgr to be used.
