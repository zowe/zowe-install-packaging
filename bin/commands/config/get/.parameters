ha-instance|i|string|||||Zowe high availability instance ID.
path|p|string|required||||Path of the configuration. For example, `components.gateway.port`.
