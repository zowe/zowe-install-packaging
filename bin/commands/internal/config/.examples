    zwe internal config get -p path-of-configuration -c /path/to/zowe.yaml

    zwe internal config set -p path-of-configuration -e new-value-of-configuration -c /path/to/zowe.yaml
    