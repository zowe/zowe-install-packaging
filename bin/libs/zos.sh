#!/bin/sh

#######################################################################
# This program and the accompanying materials are made available
# under the terms of the Eclipse Public License v2.0 which
# accompanies this distribution, and is available at
# https://www.eclipse.org/legal/epl-v20.html
#
# SPDX-License-Identifier: EPL-2.0
#
# Copyright Contributors to the Zowe Project.
#######################################################################

tso_command() {
  message="- tsocmd $@"
  print_debug "${message}"
  result=$(tsocmd "$@" 2>&1)
  code=$?
  if [ ${code} -eq 0 ]; then
    print_debug "  * Succeeded"
    print_trace "  * Exit code: ${code}"
    print_trace "  * Output:"
    if [ -n "${result}" ]; then
      print_trace "$(padding_left "${result}" "    ")"
    fi
  else
    print_debug "  * Failed"
    print_error "  * Exit code: ${code}"
    print_error "  * Output:"
    if [ -n "${result}" ]; then
      print_error "$(padding_left "${result}" "    ")"
    fi
  fi

  echo "${result}"

  return ${code}
}

operator_command() {
  cmd="${1}"

  opercmd=${ZWE_zowe_runtimeDirectory}/bin/utils/opercmd.rex

  message="- opercmd ${cmd}"
  print_debug "${message}"
  result=$("${opercmd}" "${cmd}" 2>&1)
  code=$?
  if [ ${code} -eq 0 ]; then
    print_debug "  * Succeeded"
    print_trace "  * Exit code: ${code}"
    print_trace "  * Output:"
    if [ -n "${result}" ]; then
      print_trace "$(padding_left "${result}" "    ")"
    fi
  else
    print_debug "  * Failed"
    print_error "  * Exit code: ${code}"
    print_error "  * Output:"
    if [ -n "${result}" ]; then
      print_error "$(padding_left "${result}" "    ")"
    fi
  fi

  echo "${result}"

  return ${code}
}

verify_generated_jcl() {
  jcllib=$(read_yaml "${ZWE_CLI_PARAMETER_CONFIG}" ".zowe.setup.dataset.jcllib")
  # read JCL library and validate
  does_jcl_exist=$(is_data_set_exists "${jcllib}")
  if [ -z "${does_jcl_exist}" ]; then
    result=$(zwecli_inline_execute_command init generate)
  fi

  # should be created, but may take time to discover.
  if [ -z "${does_jcl_exist}" ]; then
    does_jcl_exist=
    for secs in 1 5 10 ; do
      does_jcl_exist=$(is_data_set_exists "${jcllib}")
      if [ -z "${does_jcl_exist}" ]; then
        sleep ${secs}
      else
        break
      fi
    done

    if [ -z "${does_jcl_exist}" ]; then
      return 1
    fi
  fi
  echo "${jcllib}"
  return 0
}
