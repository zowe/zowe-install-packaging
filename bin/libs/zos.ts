/*
  This program and the accompanying materials are made available
  under the terms of the Eclipse Public License v2.0 which
  accompanies this distribution, and is available at
  https://www.eclipse.org/legal/epl-v20.html
 
  SPDX-License-Identifier: EPL-2.0
 
  Copyright Contributors to the Zowe Project.
*/

import * as std from 'cm_std';
import * as os from 'cm_os';
import * as common from './common';
import * as shell from './shell';
import * as stringlib from './string';
import * as zosDataset from './zos-dataset';
import * as initGenerate from '../commands/init/generate/index';

export function tsoCommand(...args:string[]): { rc: number, out: string } {
  let message = "tsocmd " + '"' + args.join(' ') + '"';
  common.printDebug('- '+message);
  //we echo at the end to avoid a configmgr quirk where trying to read stdout when empty can hang waiting for bytes
  const result = shell.execOutSync('sh', '-c', `${message} 2>&1 && echo '.'`);
  if (result.rc == 0) {
    common.printDebug("  * Succeeded");
    common.printTrace(`  * Exit code: ${result.rc}`);
    common.printTrace("  * Output:");
    if (result.out) {
      result.out = result.out.substring(0, result.out.length - 1);
      common.printTrace(stringlib.paddingLeft(result.out, "    "));
    }
  } else {
    common.printDebug("  * Failed");
    common.printError(`  * Exit code: ${result.rc}`);
    common.printError("  * Output:");
    if (result.out) {
      common.printError(stringlib.paddingLeft(result.out, "    "));
    }
  }
  return { rc: result.rc, out: result.out ? result.out : '' };
}

export function operatorCommand(command: string): { rc: number, out: string } {
  const opercmd=std.getenv('ZWE_zowe_runtimeDirectory')+'/bin/utils/opercmd.rex';

  let message=`- opercmd ${command}`;
  common.printDebug(message);
  //we echo at the end to avoid a configmgr quirk where trying to read stdout when empty can hang waiting for bytes
  const result = shell.execOutSync('sh', '-c', `${opercmd} "${command}" 2>&1 && echo '.'`);
  if (result.rc == 0) {
    common.printDebug("  * Succeeded");
    common.printTrace(`  * Exit code: ${result.rc}`);
    common.printTrace("  * Output:");
    if (result.out) {
      common.printTrace(stringlib.paddingLeft(result.out, "    "));
    }
  } else {
    common.printDebug("  * Failed");
    common.printError(`  * Exit code: ${result.rc}`);
    common.printError("  * Output:");
    if (result.out) {
      common.printError(stringlib.paddingLeft(result.out, "    "));
    }
  }
  //we strip the '.' we added above
  return { rc: result.rc, out: result.out ? result.out.substring(0, result.out.length-1) : '' };
}

export function verifyGeneratedJcl(config:any): string {
  const jcllib = config.zowe.setup.dataset.jcllib;
  if (!jcllib) {
    return undefined;
  }
  const expectedMember = jcllib+'(ZWEIMVS)';
  // read JCL library and validate using expected member ZWEIMVS (init mvs command)
  let doesJclExist: boolean = zosDataset.isDatasetExists(expectedMember);
  if (!doesJclExist) {
    initGenerate.execute();
  }

  // should be created, but may take time to discover.
  if (!doesJclExist) {
    const interval = [1,5,10,30];
    for (let i = 0; i < interval.length; i++) {
      let secs = interval[i];
      doesJclExist=zosDataset.isDatasetExists(expectedMember);
      if (!doesJclExist) {
        os.sleep(secs*1000);
      } else {
        break;
      }
    }

    if (!doesJclExist) {      
      return undefined;
    }
  }
  return jcllib;
}
