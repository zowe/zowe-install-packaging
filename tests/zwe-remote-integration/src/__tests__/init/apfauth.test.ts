/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */

import { REMOTE_SYSTEM_INFO } from '../../config/TestConfig';
import ZoweYamlType from '../../config/ZoweYamlType';
import * as zosfiles from '../../zos/Files';
import { RemoteTestRunner } from '../../zos/RemoteTestRunner';
import { ZoweConfig } from '../../config/ZoweConfig';
import { FileType, TestFileActions, TestFile } from '../../zos/TestFileActions';

const testSuiteName = 'init-apfauth';
describe(`${testSuiteName}`, () => {
  let testRunner: RemoteTestRunner;
  let cfgYaml: ZoweYamlType;
  let cleanupDatasets: TestFile[] = []; // a list of datasets deleted after every test

  beforeAll(() => {
    testRunner = new RemoteTestRunner(testSuiteName);
    cfgYaml = ZoweConfig.getZoweYaml(); // init
  });
  beforeEach(() => {
    cfgYaml = ZoweConfig.getZoweYaml();
  });

  afterEach(async () => {
    await testRunner.postTest();
    await TestFileActions.deleteAll(cleanupDatasets);
    cleanupDatasets = [];
  });

  afterAll(() => {
    testRunner.shutdown();
  });

  describe('(LONG)', () => {
    beforeEach(async () => {
      // re-created in every `init` subcommand based on changes to zowe yaml command...
      const jcllib: TestFile = { name: REMOTE_SYSTEM_INFO.jcllib, type: FileType.DS_NON_CLUSTER };
      // try to delete everything we know about
      await TestFileActions.deleteAll([jcllib]);
    });

    it('apf empty jcllib', async () => {
      cfgYaml.zowe.setup.dataset.jcllib = null;
      const result = await testRunner.runZweTest(cfgYaml, 'init apfauth --dry-run');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot();
      expect(result.rc).toBe(1);
    });

    it('apf empty ds prefix', async () => {
      cfgYaml.zowe.setup.dataset.prefix = null;
      const result = await testRunner.runZweTest(cfgYaml, 'init apfauth --dry-run');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot();
      expect(result.rc).toBe(1);
    });

    it('apf bad ds prefix', async () => {
      cfgYaml.zowe.setup.dataset.prefix = 'BAD.DS.PREFIX.NOEXIST';
      const result = await testRunner.runZweTest(cfgYaml, 'init apfauth --dry-run');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot();
      expect(result.rc).toBe(143);
    });
  });

  describe('(SHORT)', () => {
    beforeAll(async () => {
      cfgYaml = ZoweConfig.getZoweYaml();
      const result = await testRunner.runZweTest(cfgYaml, 'init generate --allow-overwrite');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot('short-before-all-apf');
      expect(result.rc).toBe(0);
    });
    it('apf empty jcllib post-generate', async () => {
      cfgYaml.zowe.setup.dataset.jcllib = '';
      const result = await testRunner.runZweTest(cfgYaml, 'init apfauth --dry-run');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot();
      expect(result.rc).toBe(1);
    });

    if (REMOTE_SYSTEM_INFO?.storclas?.length > 0) {
      it('apf sms-managed authLoadLib', async () => {
        const smsDs = `${REMOTE_SYSTEM_INFO.prefix}.APF.LOADLIB`;
        await zosfiles.createPds(smsDs, {
          storclass: REMOTE_SYSTEM_INFO.storclas,
        });
        cleanupDatasets.push({ name: smsDs, type: FileType.DS_NON_CLUSTER });
        cfgYaml.zowe.setup.dataset.authLoadlib = smsDs;
        const result = await testRunner.runZweTest(cfgYaml, 'init apfauth --dry-run');
        expect(result.stdout).not.toBeNull();
        expect(result.cleanedStdout).toMatchSnapshot();
        expect(result.rc).toBe(0);
      });
      it('apf sms-managed authPluginLib', async () => {
        const smsDs = `${REMOTE_SYSTEM_INFO.prefix}.APF.PLUGLIB`;
        await zosfiles.createPds(smsDs, {
          storclass: REMOTE_SYSTEM_INFO.storclas,
        });
        cleanupDatasets.push({ name: smsDs, type: FileType.DS_NON_CLUSTER });
        cfgYaml.zowe.setup.dataset.authPluginLib = smsDs;
        const result = await testRunner.runZweTest(cfgYaml, 'init apfauth --dry-run');
        expect(result.stdout).not.toBeNull();
        expect(result.cleanedStdout).toMatchSnapshot();
        expect(result.rc).toBe(0);
      });
      it('apf sms-managed authLoadLib and authPluginLib', async () => {
        const loadSmsDs = `${REMOTE_SYSTEM_INFO.prefix}.APF.LOADLIB`;
        const plugSmsDs = `${REMOTE_SYSTEM_INFO.prefix}.APF.PLUGLIB`;
        await zosfiles.createPds(loadSmsDs, {
          storclass: REMOTE_SYSTEM_INFO.storclas,
        });
        await zosfiles.createPds(plugSmsDs, {
          storclass: REMOTE_SYSTEM_INFO.storclas,
        });
        cleanupDatasets.push({ name: loadSmsDs, type: FileType.DS_NON_CLUSTER });
        cleanupDatasets.push({ name: plugSmsDs, type: FileType.DS_NON_CLUSTER });
        cfgYaml.zowe.setup.dataset.authLoadlib = loadSmsDs;
        cfgYaml.zowe.setup.dataset.authPluginLib = plugSmsDs;
        const result = await testRunner.runZweTest(cfgYaml, 'init apfauth --dry-run');
        expect(result.stdout).not.toBeNull();
        expect(result.cleanedStdout).toMatchSnapshot();
        expect(result.rc).toBe(0);
      });
    } else {
      it.skip('apf sms-managed authLoadLib', () => {});
      it.skip('apf sms-managed authPluginLib', () => {});
      it.skip('apf sms-managed authLoadLib and authPluginLib', () => {});
    }

    it('apf bad authLoadLib', async () => {
      cfgYaml.zowe.setup.dataset.authLoadlib = 'DOES.NOT.EXIST';
      const result = await testRunner.runZweTest(cfgYaml, 'init apfauth --dry-run');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot();
      expect(result.rc).toBe(64);
    });

    it('apf bad authPluginLib', async () => {
      cfgYaml.zowe.setup.dataset.authPluginLib = 'DOES.NOT.EXIST';
      const result = await testRunner.runZweTest(cfgYaml, 'init apfauth --dry-run');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot();
      expect(result.rc).toBe(64);
    });

    it('apf simple --dry-run', async () => {
      const result = await testRunner.runZweTest(cfgYaml, 'init apfauth --dry-run');
      expect(result.stdout).not.toBeNull();
      expect(result.cleanedStdout).toMatchSnapshot();
      expect(result.rc).toBe(0); // 60 is expected...
    });
  });
});
