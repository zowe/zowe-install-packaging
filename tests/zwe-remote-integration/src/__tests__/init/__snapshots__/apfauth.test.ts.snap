// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init-apfauth (LONG) apf bad ds prefix 1`] = `
"Temporary directory '/tmp/.zweenv-0000' created.
Zowe will remove it on success, but if zwe exits with a non-zero code manual cleanup would be needed.
bos extend currSize=0x0 dataSize=0x105d chunk=0x1000 extend=0x105d
-------------------------------------------------------------------------------
>> APF authorize load libraries

ERROR: ZWEL0143E Cannot find data set member 'BAD.DS.PREFIX.NOEXIST.SZWESAMP(ZWEGENER)'. You may need to re-run zwe install."
`;

exports[`init-apfauth (LONG) apf bad ds prefix: short-before-all-apf 1`] = `
"Temporary directory '/tmp/.zweenv-0000' created.
Zowe will remove it on success, but if zwe exits with a non-zero code manual cleanup would be needed.
bos extend currSize=0x0 dataSize=0x1057 chunk=0x1000 extend=0x1057
Template JCL: TEST.DATASET.PFX.SZWESAMP(ZWEGENER)
--- JCL content ---
//ZWEGENER JOB
//*
//* This program and the accompanying materials are made available
//* under the terms of the Eclipse Public License v2.0 which
//* accompanies this distribution, and is available at
//* https://www.eclipse.org/legal/epl-v20.html
//*
//* SPDX-License-Identifier: EPL-2.0
//*
//* Copyright Contributors to the Zowe Project. 2020, 2020
//*
//*********************************************************************
//*
//* This job is responsible for generating other jobs required
//* to configure Zowe.
//*
//* The method of validating your configuration is using
//* JSON Schema <https://json-schema.org>. Zowe provides
//* the ConfigMgr to assist in this. This job will invoke
//* the ConfigMgr to validate your current configuration
//* before generating any jobs. If there are any values
//* that are incorrect, you will be notified. You should
//* fix the value and then run this job again. You can run
//* this job as many times as you need.
//*
//* Configmgr documentation:
//* https://docs.zowe.org/stable/user-guide/configmgr-using
//*
//* Note: Any string with braces has an associated yaml value
//*       in one of the yaml definitions for Zowe.
//*       You must find the value and substitute it.
//*
//*       {key} -> value
//*
//GENER    EXEC PGM=IKJEFT1B
//ISPPROF  DD   DSN=,DISP=(NEW,DELETE),UNIT=,
//   DCB=(RECFM=FB,LRECL=80,BLKSIZE=3120,DSORG=PO),
//   SPACE=(3120,(20,5,10))
//*
//* Replace TEST.DATASET.PFX with the
//* Value as seen in zowe.yaml
//*
//SYSPROC  DD DSN=TEST.DATASET.PFX.SZWEEXEC,DISP=SHR
//*
//* Replace TEST.DATASET.PFX with the
//* Value as seen in zowe.yaml
//*
//STEPLIB  DD DSN=TEST.DATASET.PFX.SZWELOAD,DISP=SHR
//ISPPLIB  DD DSN=ISP.SISPPENU,DISP=SHR
//ISPMLIB  DD DSN=ISP.SISPMENU,DISP=SHR
//ISPTLIB  DD DSN=ISP.SISPTENU,DISP=SHR
//ISPSLIB  DD DSN=ISP.SISPSENU,DISP=SHR
//*
//* The order must be as follows.
//*
//* zowe-yaml-schema.json
//* server-common.json
//*
//* Replace /test/dir with where your Zowe run time
//* directory is, as seen in zowe.yaml
//*
//MYSCHEMA DD   *,DLM=$$
FILE /test/dir/schemas/zowe-yaml-schema.json
FILE /test/dir/schemas/server-common.json
$$
//*
//* The DD below must include one or more FILE or PARMLIB
//* Entries. The lower entries have their values
//* Overridden by the higher entries.
//* PARMLIB member must be named "ZWEYAML"
//*
//* Ex. PARMLIB MY.ZOWE.CUSTOM.PARMLIB
//*     FILE /some/other/zowe.yaml
//MYCONFIG DD   *,DLM=$$
FILE /test/dir/zowe.test.yaml
$$
//CMGROUT  DD   SYSOUT=*
//SYSPRINT DD   SYSOUT=*
//SYSTSPRT DD   SYSOUT=*
//*
//* Change 'generate' to 'nogenerate' if you only
//* want to validate your configuration. The default
//* option, 'generate', will validate and then generate
//* jobs based on your configuration.
//*
//*   - generate
//*   - nogenerate
//*
//* Change 'noverbose' to 'verbose' below for
//* advanced logging. This is not needed unless
//* there is an error.
//*
//*   - verbose
//*   - noverbose
//*
//SYSTSIN  DD   *
ISPSTART CMD(%ZWEGEN00 -
generate -
noverbose -
)

--- End of JCL ---
Submitting Job ZWEGENER
Job ZWEGENER(JOB00000) completed with RC=0
Zowe JCL generated successfully"
`;

exports[`init-apfauth (LONG) apf empty ds prefix 1`] = `
"Error: Validation of FILE(/test/dir/zowe.test.yaml):FILE(/test/dir/files/defaults.yaml) against schema /test/dir/schemas/zowe-yaml-schema.json:/test/dir/schemas/server-common.json found invalid JSON Schema data
Validity Exceptions(s) with object at 
  Validity Exceptions(s) with object at /zowe
    Validity Exceptions(s) with object at /zowe/setup
      Validity Exceptions(s) with object at /zowe/setup/dataset
        type 'null' not permitted at /zowe/setup/dataset/prefix; expecting type 'string'"
`;

exports[`init-apfauth (LONG) apf empty jcllib 1`] = `
"Error: Validation of FILE(/test/dir/zowe.test.yaml):FILE(/test/dir/files/defaults.yaml) against schema /test/dir/schemas/zowe-yaml-schema.json:/test/dir/schemas/server-common.json found invalid JSON Schema data
Validity Exceptions(s) with object at 
  Validity Exceptions(s) with object at /zowe
    Validity Exceptions(s) with object at /zowe/setup
      Validity Exceptions(s) with object at /zowe/setup/dataset
        type 'null' not permitted at /zowe/setup/dataset/jcllib; expecting type 'string'"
`;

exports[`init-apfauth (SHORT) apf bad authLoadLib 1`] = `
"Temporary directory '/tmp/.zweenv-0000' created.
Zowe will remove it on success, but if zwe exits with a non-zero code manual cleanup would be needed.
bos extend currSize=0x0 dataSize=0x104d chunk=0x1000 extend=0x104d
-------------------------------------------------------------------------------
>> APF authorize load libraries

ERROR:   * Exit code: 8
ERROR:   * Output:
ERROR:     listds 'DOES.NOT.EXIST' label
    DOES.NOT.EXIST
    IKJ58503I DATA SET 'DOES.NOT.EXIST' NOT IN CATALOG
ERROR: Error ZWEL0320E: The dataset specified in 'zowe.setup.dataset.authLoadlib' does not exist."
`;

exports[`init-apfauth (SHORT) apf bad authPluginLib 1`] = `
"Temporary directory '/tmp/.zweenv-0000' created.
Zowe will remove it on success, but if zwe exits with a non-zero code manual cleanup would be needed.
bos extend currSize=0x0 dataSize=0x104e chunk=0x1000 extend=0x104e
-------------------------------------------------------------------------------
>> APF authorize load libraries

ERROR:   * Exit code: 8
ERROR:   * Output:
ERROR:     listds 'DOES.NOT.EXIST' label
    DOES.NOT.EXIST
    IKJ58503I DATA SET 'DOES.NOT.EXIST' NOT IN CATALOG
ERROR: Error ZWEL0320E: The dataset specified in 'zowe.setup.dataset.authPluginLib' does not exist."
`;

exports[`init-apfauth (SHORT) apf empty jcllib post-generate 1`] = `
"Error: Validation of FILE(/test/dir/zowe.test.yaml):FILE(/test/dir/files/defaults.yaml) against schema /test/dir/schemas/zowe-yaml-schema.json:/test/dir/schemas/server-common.json found invalid JSON Schema data
Validity Exceptions(s) with object at 
  Validity Exceptions(s) with object at /zowe
    Validity Exceptions(s) with object at /zowe/setup
      Validity Exceptions(s) with object at /zowe/setup/dataset
        Validity Exceptions(s) with string at /zowe/setup/dataset/jcllib
          string too short (len=0) '' 0 < MIN=3 at /zowe/setup/dataset/jcllib
          string pattern match fail s='', pat='^([A-Z\\$\\#\\@]){1}([A-Z0-9\\$\\#\\@\\-]){0,7}(\\.([A-Z\\$\\#\\@]){1}([A-Z0-9\\$\\#\\@\\-]){0,7}){0,11}$', at /zowe/setup/dataset/jcllib"
`;

exports[`init-apfauth (SHORT) apf simple --dry-run 1`] = `
"Temporary directory '/tmp/.zweenv-0000' created.
Zowe will remove it on success, but if zwe exits with a non-zero code manual cleanup would be needed.
bos extend currSize=0x0 dataSize=0x1057 chunk=0x1000 extend=0x1057
-------------------------------------------------------------------------------
>> APF authorize load libraries

Template JCL: TEST.DATASET.PFX.SZWESAMP(ZWEIAPF2) , Executable JCL: TEST.DATASET.PFX.JCLLIB(ZWEIAPF2)
--- JCL Content ---
//ZWEIAPF2 JOB
//*
//* This program and the accompanying materials are made available
//* under the terms of the Eclipse Public License v2.0 which
//* accompanies this distribution, and is available at
//* https://www.eclipse.org/legal/epl-v20.html
//*
//* SPDX-License-Identifier: EPL-2.0
//*
//* Copyright Contributors to the Zowe Project. 2020, 2020
//*
//*********************************************************************
//*
//*  This JCL is used to set APF for the two datasets of Zowe
//*  Which need it. You can issue this, or use another
//*  Way to accomplish the task.
//*
//*  The following variables are derived from the zowe YAML config:
//*  LOADLIB: the dataset that holds the APF portion of Zowe
//*  PLUGLIB: The dataset that holds the extensions for ZIS.
//*
//*********************************************************************
//*
//APFAUTH EXEC PGM=BPXBATCH
//BPXPRINT DD SYSOUT=*
//STDOUT   DD SYSOUT=*
//STDERR   DD SYSOUT=*
//STDPARM DD *
SH cd '/test/dir' &&
cd bin/utils &&
export LOADLIB='TEST.DATASET.PFX.SZWELOAD' &&
export LOADLOC="VOLUME=TSTVOL" &&
export PLUGLIB='TEST.DATASET.PFX.ZWESAPL' &&
export PLUGLOC="VOLUME=TSTVOL" &&
./opercmd.rex "SETPROG APF,ADD,DSN=$LOADLIB,$LOADLOC" &&
./opercmd.rex "SETPROG APF,ADD,DSN=$PLUGLIB,$PLUGLOC"
//*
--- End of JCL ---
JCL not submitted, command run with "--dry-run" flag.
To perform command, re-run command without "--dry-run" flag, or submit the JCL directly

>> Command run successfully.


>> Zowe load libraries are APF authorized successfully."
`;

exports[`init-apfauth (SHORT) apf sms-managed authLoadLib 1`] = `
"Temporary directory '/tmp/.zweenv-0000' created.
Zowe will remove it on success, but if zwe exits with a non-zero code manual cleanup would be needed.
bos extend currSize=0x0 dataSize=0x105a chunk=0x1000 extend=0x105a
-------------------------------------------------------------------------------
>> APF authorize load libraries

Template JCL: TEST.DATASET.PFX.SZWESAMP(ZWEIAPF2) , Executable JCL: TEST.DATASET.PFX.JCLLIB(ZWEIAPF2)
--- JCL Content ---
//ZWEIAPF2 JOB
//*
//* This program and the accompanying materials are made available
//* under the terms of the Eclipse Public License v2.0 which
//* accompanies this distribution, and is available at
//* https://www.eclipse.org/legal/epl-v20.html
//*
//* SPDX-License-Identifier: EPL-2.0
//*
//* Copyright Contributors to the Zowe Project. 2020, 2020
//*
//*********************************************************************
//*
//*  This JCL is used to set APF for the two datasets of Zowe
//*  Which need it. You can issue this, or use another
//*  Way to accomplish the task.
//*
//*  The following variables are derived from the zowe YAML config:
//*  LOADLIB: the dataset that holds the APF portion of Zowe
//*  PLUGLIB: The dataset that holds the extensions for ZIS.
//*
//*********************************************************************
//*
//APFAUTH EXEC PGM=BPXBATCH
//BPXPRINT DD SYSOUT=*
//STDOUT   DD SYSOUT=*
//STDERR   DD SYSOUT=*
//STDPARM DD *
SH cd '/test/dir' &&
cd bin/utils &&
export LOADLIB='TEST.DATASET.PFX.SZWELOAD' &&
export LOADLOC=SMS &&
export PLUGLIB='TEST.DATASET.PFX.ZWESAPL' &&
export PLUGLOC="VOLUME=TSTVOL" &&
./opercmd.rex "SETPROG APF,ADD,DSN=$LOADLIB,$LOADLOC" &&
./opercmd.rex "SETPROG APF,ADD,DSN=$PLUGLIB,$PLUGLOC"
//*
--- End of JCL ---
JCL not submitted, command run with "--dry-run" flag.
To perform command, re-run command without "--dry-run" flag, or submit the JCL directly

>> Command run successfully.


>> Zowe load libraries are APF authorized successfully."
`;

exports[`init-apfauth (SHORT) apf sms-managed authLoadLib and authPluginLib 1`] = `
"Temporary directory '/tmp/.zweenv-0000' created.
Zowe will remove it on success, but if zwe exits with a non-zero code manual cleanup would be needed.
bos extend currSize=0x0 dataSize=0x105e chunk=0x1000 extend=0x105e
-------------------------------------------------------------------------------
>> APF authorize load libraries

Template JCL: TEST.DATASET.PFX.SZWESAMP(ZWEIAPF2) , Executable JCL: TEST.DATASET.PFX.JCLLIB(ZWEIAPF2)
--- JCL Content ---
//ZWEIAPF2 JOB
//*
//* This program and the accompanying materials are made available
//* under the terms of the Eclipse Public License v2.0 which
//* accompanies this distribution, and is available at
//* https://www.eclipse.org/legal/epl-v20.html
//*
//* SPDX-License-Identifier: EPL-2.0
//*
//* Copyright Contributors to the Zowe Project. 2020, 2020
//*
//*********************************************************************
//*
//*  This JCL is used to set APF for the two datasets of Zowe
//*  Which need it. You can issue this, or use another
//*  Way to accomplish the task.
//*
//*  The following variables are derived from the zowe YAML config:
//*  LOADLIB: the dataset that holds the APF portion of Zowe
//*  PLUGLIB: The dataset that holds the extensions for ZIS.
//*
//*********************************************************************
//*
//APFAUTH EXEC PGM=BPXBATCH
//BPXPRINT DD SYSOUT=*
//STDOUT   DD SYSOUT=*
//STDERR   DD SYSOUT=*
//STDPARM DD *
SH cd '/test/dir' &&
cd bin/utils &&
export LOADLIB='TEST.DATASET.PFX.SZWELOAD' &&
export LOADLOC=SMS &&
export PLUGLIB='TEST.DATASET.PFX.ZWESAPL' &&
export PLUGLOC=SMS &&
./opercmd.rex "SETPROG APF,ADD,DSN=$LOADLIB,$LOADLOC" &&
./opercmd.rex "SETPROG APF,ADD,DSN=$PLUGLIB,$PLUGLOC"
//*
--- End of JCL ---
JCL not submitted, command run with "--dry-run" flag.
To perform command, re-run command without "--dry-run" flag, or submit the JCL directly

>> Command run successfully.


>> Zowe load libraries are APF authorized successfully."
`;

exports[`init-apfauth (SHORT) apf sms-managed authPluginLib 1`] = `
"Temporary directory '/tmp/.zweenv-0000' created.
Zowe will remove it on success, but if zwe exits with a non-zero code manual cleanup would be needed.
bos extend currSize=0x0 dataSize=0x105b chunk=0x1000 extend=0x105b
-------------------------------------------------------------------------------
>> APF authorize load libraries

Template JCL: TEST.DATASET.PFX.SZWESAMP(ZWEIAPF2) , Executable JCL: TEST.DATASET.PFX.JCLLIB(ZWEIAPF2)
--- JCL Content ---
//ZWEIAPF2 JOB
//*
//* This program and the accompanying materials are made available
//* under the terms of the Eclipse Public License v2.0 which
//* accompanies this distribution, and is available at
//* https://www.eclipse.org/legal/epl-v20.html
//*
//* SPDX-License-Identifier: EPL-2.0
//*
//* Copyright Contributors to the Zowe Project. 2020, 2020
//*
//*********************************************************************
//*
//*  This JCL is used to set APF for the two datasets of Zowe
//*  Which need it. You can issue this, or use another
//*  Way to accomplish the task.
//*
//*  The following variables are derived from the zowe YAML config:
//*  LOADLIB: the dataset that holds the APF portion of Zowe
//*  PLUGLIB: The dataset that holds the extensions for ZIS.
//*
//*********************************************************************
//*
//APFAUTH EXEC PGM=BPXBATCH
//BPXPRINT DD SYSOUT=*
//STDOUT   DD SYSOUT=*
//STDERR   DD SYSOUT=*
//STDPARM DD *
SH cd '/test/dir' &&
cd bin/utils &&
export LOADLIB='TEST.DATASET.PFX.SZWELOAD' &&
export LOADLOC="VOLUME=TSTVOL" &&
export PLUGLIB='TEST.DATASET.PFX.ZWESAPL' &&
export PLUGLOC=SMS &&
./opercmd.rex "SETPROG APF,ADD,DSN=$LOADLIB,$LOADLOC" &&
./opercmd.rex "SETPROG APF,ADD,DSN=$PLUGLIB,$PLUGLOC"
//*
--- End of JCL ---
JCL not submitted, command run with "--dry-run" flag.
To perform command, re-run command without "--dry-run" flag, or submit the JCL directly

>> Command run successfully.


>> Zowe load libraries are APF authorized successfully."
`;
